gnovm/cmd/gno/clean_test.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCleanApp(t *testing.T) {
+	tc := []testMainCase{
+		{
+			args:        []string{"clean -h"},
+			errShouldBe: "flag: help requested",
+		},
+		{
+			args:        []string{"clean unknown"},
+			errShouldBe: "flag: help requested",
+		},
+		{
+			args:                 []string{"clean"},
+			testDir:              "../../tests/integ/empty-dir",
+			simulateExternalRepo: true,
+			errShouldBe:          "not a gno module",
+		},
+		{
+			args:                 []string{"clean"},
+			testDir:              "../../tests/integ/minimalist-gnomod",
+			simulateExternalRepo: true,
+		},
+	}
+	testMainCaseRun(t, tc)
+
+	workingDir, err := os.Getwd()
+	require.Nil(t, err)
+
+	// Test clean command
+	for _, tc := range []struct {
+		desc         string
+		args         []string
+		files        []string
+		filesRemoved []string
+		stdOut       string
+	}{
+		{
+			desc:         "only_generated_files",
+			args:         []string{"clean"},
+			files:        []string{"gno.mod", "tmp.gno.gen.go", "tmp.gno.gen_test.go"},
+			filesRemoved: []string{"tmp.gno.gen.go", "tmp.gno.gen_test.go"},
+			stdOut:       "rm tmp.gno.gen.go\nrm tmp.gno.gen_test.go\n",
+		},
+		{
+			desc:  "no_generated_files",
+			args:  []string{"clean"},
+			files: []string{"gno.mod", "tmp.gno"},
+		},
+		{
+			desc:         "mixed_files",
+			args:         []string{"clean"},
+			files:        []string{"gno.mod", "README.md", "tmp.gno", "tmp_test.gno", "tmp.gno.gen.go", "tmp.gno.gen_test.go"},
+			filesRemoved: []string{"tmp.gno.gen.go", "tmp.gno.gen_test.go"},
+			stdOut:       "rm tmp.gno.gen.go\nrm tmp.gno.gen_test.go\n",
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			mockOut := bytes.NewBufferString("")
+			mockErr := bytes.NewBufferString("")
+
+			// test clean with external repo
+			tmpDir, cleanUpFn := createTmpDir(t)
+			defer cleanUpFn()
+
+			// cd to tmp directory
+			os.Chdir(tmpDir)
+			defer os.Chdir(workingDir)
+
+			// create files
+			for _, file := range tc.files {
+				err = os.WriteFile(file, []byte("test"), 0o644)
+				require.Nil(t, err)
+			}
+
+			// set up io
+			io := commands.NewTestIO()
+			io.SetOut(commands.WriteNopCloser(mockOut))
+			io.SetErr(commands.WriteNopCloser(mockErr))
+
+			// dry run clean
+			err = newGnodevCmd(io).ParseAndRun(context.Background(), []string{"clean", "-n"})
+			require.Nil(t, err)
+			// check output
+			if tc.stdOut != "" {
+				assert.Equal(t, tc.stdOut, mockOut.String())
+			}
+			// check files
+			for _, file := range tc.files {
+				_, err = os.Stat(file)
+				require.Nil(t, err)
+			}
+
+			// run clean
+			err = newGnodevCmd(io).ParseAndRun(context.Background(), []string{"clean"})
```

Comment:
How about switching from 'Gnodev' to 'Gnocli' in our issues for clarity?

Reply:
I think using "gnocli" instead of "gnodev" in issues is a good idea.

Should we also rename `s/newGnodevCmd/newGnoCmd`? or something better.

Reply:
`s/newGnodevCmd/newGnocliCmd/g`, please.